File: codeforces/low_difficulty/800_points/71.plus_one_on_the_subset_[1624A]/poots.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned int t, n;
  long int a, major, minor;
  cin >> t;
  while (t--) {
    cin >> n;
    cin >> a;
    major = minor = a;
    n--;
    while (n--) {
      cin >> a;
      if (major < a) {
        major = a;
      }
      if (minor > a) {
        minor = a;
      }
    }
    cout << (major - minor) << endl;
  }
  return 0;
}

File: codeforces/low_difficulty/difficulty_800/71.plus_one_on_the_subset_[1624A]/poots.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n = int(input())
  a = list(map(int, input().split(" ")))
  major = minor = a[0]
  for i in range(1, n):
    if (major < a[i]):
      major = a[i]
    if (minor > a[i]):
      minor = a[i]
  print(major - minor)
  t -= 1